federation/pkg/dnsprovider/providers/aws/route53/stubs/route53api.go

```diff
@@ -108,9 +115,28 @@ func (r *Route53APIStub) ListHostedZonesPages(input *route53.ListHostedZonesInpu
 	return nil
 }
 
+func (r *Route53APIStub) ListHostedZonesByName(input *route53.ListHostedZonesByNameInput) (*route53.ListHostedZonesByNameOutput, error) {
+	output := &route53.ListHostedZonesByNameOutput{}
+	for _, zone := range r.zones {
+		if strings.TrimSuffix(aws.StringValue(input.DNSName), ".") == strings.TrimSuffix(aws.StringValue(zone.Name), ".") {
+			output.HostedZones = append(output.HostedZones, zone)
+		}
+	}
+	return output, nil
+}
+
+func (r *Route53APIStub) GetHostedZone(input *route53.GetHostedZoneInput) (*route53.GetHostedZoneOutput, error) {
+	id := "/hostedzone/" + aws.StringValue(input.Id)
+	zone, ok := r.zones[id]
+	if !ok {
+		return nil, fmt.Errorf("Error getting hosted DNS zone: %s does not exists", id)
+	}
+	return &route53.GetHostedZoneOutput{HostedZone: zone}, nil
+}
+
 func (r *Route53APIStub) CreateHostedZone(input *route53.CreateHostedZoneInput) (*route53.CreateHostedZoneOutput, error) {
 	name := aws.StringValue(input.Name)
-	id := "/hostedzone/" + name
+	id := "/hostedzone/" + string(uuid.NewUUID())
```

Comment:
Both are correct. I changed this because in AWS the id is not the same as the DNS name. I don't want test pass because the DNS name happens to be same as id